webpack/components/AnsibleRoles/components/AnsibleRolesImportButtonWrapper.js

```diff
@@ -0,0 +1,27 @@
+import React from 'react';
+import { translate as __ } from 'foremanReact/common/I18n';
+import { useAPI } from 'foremanReact/common/hooks/API/APIHooks';
+import { showToast } from '../../../toastHelper';
+import { AnsibleRolesImportButton } from './AnsibleRolesImportButton';
+
+export const AnsibleRolesImportButtonWrapper = () => {
```

Comment:
This is an interesting approach. We don't have this in Foreman, but now I'm wondering if this should be a common pattern. For example, provide a `hasPermission(permission)` function.

Linking back to my earlier approach, another one could be that you return the URL in the API response if the user has permissions. If they don't, then you can derive that in the frontend without checking permissions manually.

Reply:
Theres a discussion on permissions here: https://community.theforeman.org/t/handling-permissions-whats-our-approach/36858 BTW

Reply:
Oh yes, I almost forgot about that post. I actually found a solution to the permissions problem that I think holds promise and am currently implementing a proof of concept.